packages/server/src/main.rs

```diff
@@ -209,20 +206,93 @@ async fn handle_events(
+        );
+    }
+    // Get first GPU as default for encoding
+    let gpu = gpus.first().unwrap();
+    println!("Selecting first GPU: '{}'", gpu.device_name());
+
+    println!("Getting compatible encoders..");
+    let encoders = enc_helper::get_compatible_encoders();
+    for encoder in &encoders {
+        println!(
+            "> [Encoder] Name: '{}', Codec: '{}', API: '{}', Type: '{}'",
+            encoder.name,
+            encoder.codec.to_str(),
+            encoder.encoder_api.to_str(),
+            encoder.encoder_type.to_str()
+        );
+    }
+    // Pick most suitable H.264 encoder (TODO: Codec priority/scoring)
+    let mut encoder = enc_helper::get_best_compatible_encoder(
+        &encoders,
+        enc_helper::VideoCodec::H264,
+        enc_helper::EncoderType::HARDWARE,
+    );
+    if encoder.is_none() {
+        println!("No suitable hardware encoder found for H.264 codec. Trying software encoder..");
+        encoder = enc_helper::get_best_compatible_encoder(
+            &encoders,
+            enc_helper::VideoCodec::H264,
+            enc_helper::EncoderType::SOFTWARE,
+        );
+        if encoder.is_none() {
+            println!("No suitable software encoder found for H.264 codec. Exiting as we cannot encode a stream..");
+            return Ok(());
+        }
+    }
+    let encoder = encoder.unwrap();
+    println!(
+        "Selected encoder: '{}', Codec: '{}', API: '{}', Type: '{}'",
+        encoder.name,
+        encoder.codec.to_str(),
+        encoder.encoder_api.to_str(),
+        encoder.encoder_type.to_str()
+    );
+
+    let cqp_default = 25;
+    println!("Optimizing encoder parameters with CQP of: {}", cqp_default);
+    let optimized_encoder = enc_helper::encoder_low_latency_cqp_params(&encoder, cqp_default);
+    if optimized_encoder.is_none() {
+        println!("Failed to optimize encoder parameters. Unsupported encoder? Exiting..");
+        return Ok(());
+    }
+    let optimized_encoder = optimized_encoder.unwrap();
+    println!(
+        "Optimized encoder parameters: '{}'",
+        optimized_encoder.get_parameters_string()
+    );
+
+    // Construct the pipeline arguments with the encoder
+    // TODO: AV1 codec support, resolution, framerate, etc.
     let pipeline = gst::parse::launch(
-        "
+        format!(
+            "
         waylanddisplaysrc \
-        ! video/x-raw,width=1280,height=720,format=RGBx,framerate=60/1 \
-        ! videoconvertscale\
-        ! qsvh264enc low-latency=true \
+        ! video/x-raw,width=1280,height=720,framerate=60/1,format=RGBx \
+        ! tee name=src \
+        src. ! queue2 max-size-time=2000000 ! videoconvert \
+        ! {} {} \
         ! h264parse \
-        ! isofmp4mux name=mux chunk-duration=1 fragment-duration=1 \
-        ! moqsink url=https://relay.fst.so namespace=testing
+        ! isofmp4mux chunk-duration=1 fragment-duration=1 ! sinker. \
+        moqsink url=https://relay.dathorse.com:8443 path=testhorse name=sinker
```

Comment:
Would it make sense to have a configuration parameters for the relay endpoint? I think now could be a good point to make it configurable.

Reply:
Yeah, it would be good to have that + resolution, framerate etc. - do you think passing those as "--parameter value" to the program would be good enough?

Or I could do as I do in my personal Go projects, where:
1. defaults are used if nothing is given (activating dev environment usually)
2. ENV variables are used if given over defaults
3. given program parameters `--parameter value` are used, taking priority over ENV variables if both are set

Reply:
Sounds perfect 👌

Reply:
@AquaWolf New commit done, took suprising amount of time just to get some parameters going, but atleast there's bunch now :smiley:
 
 If the changes solve the issue here, please mark as resolved, any additional comments should be made into new conversation for new commit :+1:

Reply:
Looks good. Not sure how to solve the issue because I didn't created one.